pkg/services/accesscontrol/accesscontrol.go

```diff
@@ -120,3 +124,39 @@ func ValidateScope(scope string) bool {
 	}
 	return !strings.ContainsAny(prefix, "*?")
 }
+
+func addActionToMetadata(allMetadata map[string]Metadata, action, id string) map[string]Metadata {
+	metadata, initialized := allMetadata[id]
+	if !initialized {
+		metadata = Metadata{action: true}
+	} else {
+		metadata[action] = true
+	}
+	allMetadata[id] = metadata
+	return allMetadata
+}
+
+// GetResourcesMetadata returns a map of accesscontrol metadata, listing for each resource, users available actions
+func GetResourcesMetadata(ctx context.Context, permissions []*Permission, resource string, resourceIDs map[string]bool) (map[string]Metadata, error) {
```

Comment:
They api would look nicer if we take an slice of resourceIDs and create the lookup map inside instead of relying on the caller to create the map

Reply:
I only have in mind the use case of the caller building the id list from a list of resources. This change would imply one loop on the list of resources, then running a second loop on the list of ids to build the map, instead of running only once to build the map straight away. Do you think the cost of an additional loop is still worth implementing?